llpc/builder/llpcBuilderImpl.h

```diff
@@ -438,6 +438,169 @@ class BuilderImplImage : virtual public BuilderImplBase
     LLPC_DISALLOW_DEFAULT_CTOR(BuilderImplImage)
     LLPC_DISALLOW_COPY_AND_ASSIGN(BuilderImplImage)
 
+    // Represents the type of sampler filter.
+    enum class SamplerFilter : uint32_t
```

Comment:
Why do we need 32 bits for 2 possible values? Will this enum or the enums below be serialized?

Reply:
Ok, I'll use bool instead for 2 possible value cases.

Reply:
Enum is fine.
I was trying to point out that you force all the enums below to be 32-bits wide without explaining why you need it. By default, I'd expect you to leave out the type part and let a compiler pick something. Forcing a particular type would be understandable if you had to serialize these enums or perform frequent int-to-enum/enum-to-int conversions, but in that case it may make more sense to use `std::underlying_type<SomeEnum>::type` to get the right integer type instead.